rhp/v4/validation.go

```diff
@@ -149,6 +151,38 @@ func (req *RPCRenewContractRequest) Validate(pk types.PublicKey, tip types.Chain
 	}
 }
 
+// Validate validates a renew contract request. Prices are not validated
+func (req *RPCRefreshContractRequest) Validate(pk types.PublicKey, expirationHeight uint64, maxCollateral types.Currency) error {
+	if err := req.Prices.Validate(pk); err != nil {
+		return fmt.Errorf("prices are invalid: %w", err)
+	}
+
+	// validate the request fields
+	switch {
+	case req.MinerFee.IsZero():
+		return errors.New("miner fee must be greater than 0")
+	case req.Basis == (types.ChainIndex{}):
+		return errors.New("basis must be set")
+	}
+
+	// validate the contract fields
+	hp := req.Prices
+	// calculate the minimum allowance required for the contract based on the
+	// host's locked collateral and the contract duration
+	minRenterAllowance := MinRenterAllowance(hp, expirationHeight-req.Prices.TipHeight, req.Refresh.Collateral)
+
+	switch {
+	case req.Refresh.Allowance.IsZero():
+		return errors.New("allowance is zero")
+	case req.Refresh.Collateral.Cmp(maxCollateral) > 0:
+		return fmt.Errorf("collateral %v exceeds max collateral %v", req.Refresh.Collateral, maxCollateral)
+	case req.Refresh.Allowance.Cmp(minRenterAllowance) < 0:
+		return fmt.Errorf("allowance %v is less than minimum %v for collateral", req.Refresh.Allowance, minRenterAllowance)
```

Comment:
I'm guessing "for collateral" is a typo in the error message and it's meant to be "allowance x is less than min allowance y". It's repeated three times in the repo.